src/python/pants/jvm/goals/coursier.py

```diff
@@ -57,38 +57,35 @@ class CoursierResolve(Goal):
     subsystem_cls = CoursierResolveSubsystem
 
 
-class JvmResolvesToAddresses(FrozenDict[str, FrozenOrderedSet[Address]]):
+class JvmResolvesToArtifacts(FrozenDict[str, FrozenOrderedSet[Address]]):
     pass
 
 
 @rule
 async def map_resolves_to_consuming_targets(
     all_targets: AllTargets, jvm: JvmSubsystem
-) -> JvmResolvesToAddresses:
-    resolve_to_addresses = defaultdict(set)
+) -> JvmResolvesToArtifacts:
+    resolve_to_artifacts = defaultdict(set)
     for tgt in all_targets:
-        if tgt.has_field(JvmResolveField):
-            resolve = tgt[JvmResolveField].value or jvm.default_resolve
-            # TODO: remove once we always have a default resolve.
-            if not resolve:
-                continue
-            resolve_to_addresses[resolve].add(tgt.address)
-        elif tgt.has_field(JvmCompatibleResolvesField):
-            # TODO: add a `default_compatible_resolves` field.
-            resolves: Sequence[str] = tgt[JvmCompatibleResolvesField].value or (
-                [jvm.default_resolve] if jvm.default_resolve is not None else []
-            )
-            for resolve in resolves:
-                resolve_to_addresses[resolve].add(tgt.address)
-    return JvmResolvesToAddresses(
-        (resolve, FrozenOrderedSet(addresses))
-        for resolve, addresses in resolve_to_addresses.items()
+        if not tgt.has_field(JvmArtifactCompatibleResolvesField):
+            continue
+        # TODO: add a `default_compatible_resolves` field.
+        # TODO: error if no resolves for the target? Which would mean it's practically invisible.
```

Comment:
This feels like a bug waiting to happen. I think we should error.

FYI the fix will be to either make sure `[jvm].default_compatible_resolves` is set or to manually set it for the target.

Reply:
https://github.com/pantsbuild/pants/blob/18350c5f6f94b889639990034be9a6f771df67e0/src/python/pants/jvm/subsystems.py#L23-L29 is relevant here, and could probably be used now.

And as previously mentioned, I think that we should set an actual default value there.

Reply:
This is already using `--default-resolve` if you look at the next line.

>  I think that we should set an actual default value there.

Agreed, waiting to do in a followup.